server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala

```diff
@@ -47,3 +48,44 @@ class S3FileSigner(
     s3Client.generatePresignedUrl(request)
   }
 }
+
+class AzureFileSigner(name: URI,
+                      accountName: String,
+                      storageKey: String,
+                      preSignedUrlTimeoutSeconds: Long) extends CloudFileSigner {
+
+  private def getCloudStorageAccount: CloudStorageAccount = {
+    val connectionString = Seq(
+      "DefaultEndpointsProtocol=https",
+      s"AccountName=$accountName",
+      s"AccountKey=$storageKey",
+      "EndpointSuffix=core.windows.net"
+    ).mkString(";")
+    CloudStorageAccount.parse(connectionString)
+  }
+
+  private val cloudStorageAccount = getCloudStorageAccount
+
+  override def sign(bucket: String, objectKey: String): URL = {
+
+    val startDate = new Date(System.currentTimeMillis())
+    val expirationDate =
+      new Date(System.currentTimeMillis() + SECONDS.toMillis(preSignedUrlTimeoutSeconds))
+
+    // Generate a Shared Access Policy with expiration
+    val sharedAccessAccountPolicy = new SharedAccessAccountPolicy()
+    sharedAccessAccountPolicy.setPermissionsFromString("racwdlup")
+    sharedAccessAccountPolicy.setSharedAccessStartTime(startDate)
+    sharedAccessAccountPolicy.setSharedAccessExpiryTime(expirationDate)
+    sharedAccessAccountPolicy.setResourceTypeFromString("sco")
+    sharedAccessAccountPolicy.setServiceFromString("bfqt")
+
+    val sasToken = cloudStorageAccount.generateSharedAccessSignature(sharedAccessAccountPolicy)
```

Comment:
Is `sasToken` only allowed to access the given `bucket` and `objectKey`?  I don't see any place passing them to generate `sasToken`. Looks like `sasToken` has the same power as the long live storage key.